presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveCoercion.java

```diff
@@ -110,6 +110,9 @@
                         "    smallint_to_bigint         SMALLINT," +
                         "    int_to_bigint              INT," +
                         "    bigint_to_varchar          BIGINT," +
+                        "    float_to_varchar           FLOAT," +
+                        "    double_to_varchar          DOUBLE," +
+                        "    decimal_to_varchar         DECIMAL(10,2)," +
```

Comment:
Should we also test if something like `123456.12` as `DECIMAL(10, 4)` gets coerced to `123456.1200` when converted to string? Or is `123456.12` fine?

FWIW Hive 3.1 coerces it to `123456.12`.

Reply:
> Or is `123456.12` fine?

it's fine. 

We should add a test with overflow though.

Reply:
Sure, will add one.

Reply:
I've added one to verify that truncation is happening for coercion from `DECIMAL(10, 2)` to `VARCHAR(5)`. The test is written to ensure behaviour matches Hive 3.1.

Reply:
Since your code is not doing truncation explicitly, I think it's going to return incorrect data into the engine.
It may be hidden from client's eyes somehow, but you can also test with `... WHERE decimal_to_varchar = 'truncated_value' OR rand() = 42`.

Reply:
Thanks, let me build locally and verify.

Reply:
@findepi You are correct. My assumption that a Varchar(5) wouldn't allow longer strings to be written seems to be wrong given the latest test run. I'm unsure here how should this be handled? I wasn't able to find any similar code to learn from. And the `IntegerNumberToVarchar` coercion also seems to have the same "bug".

java.lang.AssertionError: Could not find rows:
2020-08-26T12:56:17.9317041Z tests               | [-1, 2, -3, 100, -101, 2323, 12345, 0.5, 0.5, 12345678.12, **12345**, 0.50000000000, 12345678.1200, 12345678.1200, 12345678.12, 12345678.12345612345600, 12345.12402343750, 12345.12345000000, 1]
2020-08-26T12:56:17.9317872Z tests               | [1, -2, null, -100, 101, -2323, -12345, -1.5, -1.5, -12345678.12, **-1234**, -1.50000000000, -12345678.1200, -12345678.1200, -12345678.12, -12345678.12345612345600, -12345.12402343750, -12345.12345000000, 1]
2020-08-26T12:56:17.9318033Z tests               | 
2020-08-26T12:56:17.9318171Z tests               | actual rows:
2020-08-26T12:56:17.9318674Z tests               | [-1, 2, -3, 100, -101, 2323, 12345, 0.5, 0.5, 12345678.12, **12345678.12**, 0.50000000000, 12345678.1200, 12345678.1200, 12345678.12, 12345678.12345612345600, 12345.12402343750, 12345.12345000000, 1]
2020-08-26T12:56:17.9319472Z tests               | [1, -2, null, -100, 101, -2323, -12345, -1.5, -1.5, -12345678.12, **-12345678.12**, -1.50000000000, -12345678.1200, -12345678.1200, -12345678.12, -12345678.12345612345600, -12345.12402343750, -12345.12345000000, 1]

Reply:
> And the IntegerNumberToVarchar coercion also seems to have the same "bug".

Can you please file an issue for this?

Also, when truncation would occur, what is hive's behavior?
fail / truncate / null-out / anything else?

cc @martint @electrum @losipiuk

Reply:
For any number type to  VARCHAR(n) Hive 3.1 keeps leftmost n characters and truncates the rest.

i.e. -1.1 to VARCHAR(1) becomes "-"

Will create an issue for the IntegerToVarchar coercer as well.